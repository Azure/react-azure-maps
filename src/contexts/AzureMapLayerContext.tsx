import React, { createContext } from 'react'
import { IAzureLayerStatefulProviderProps, IAzureMapLayerProps } from '../types'
import { useAzureMapLayer } from '../hooks/useAzureMapLayer'

const AzureMapLayerContext = createContext<IAzureMapLayerProps>({
  layerRef: null
})
const { Provider, Consumer: AzureMapLayerConsumer } = AzureMapLayerContext

const AzureMapLayerStatefulProvider = ({
  id,
  options,
  type,
  events,
  lifecycleEvents,
  onCreateCustomLayer
}: IAzureLayerStatefulProviderProps) => {
  const { layerRef } = useAzureMapLayer({
    id,
    options,
    type,
    events,
    lifecycleEvents,
    onCreateCustomLayer
  })
  return (
    <Provider
      value={{
        layerRef
      }}
    ></Provider>
  )
}

export {
  AzureMapLayerContext,
  AzureMapLayerConsumer,
  AzureMapLayerStatefulProvider as AzureMapLayerProvider
}
