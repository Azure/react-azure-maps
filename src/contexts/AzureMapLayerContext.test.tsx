import React from 'react'
import { render, act } from '@testing-library/react'
import { AzureMapLayerProvider } from './AzureMapLayerContext'

describe('AzureMapLayerProvider', () => {
  it('should create and render AzureMapLayerProvider', () => {
    const { container } = render(<AzureMapLayerProvider type="SymbolLayer" />)
    expect(container).toMatchSnapshot()
  })
})
