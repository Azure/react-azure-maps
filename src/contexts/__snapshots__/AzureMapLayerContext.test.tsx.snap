// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureMapLayerProvider should create and render AzureMapLayerProvider 1`] = `<div />`;
